core/go/internal/groupmgr/manager.go

```diff
@@ -0,0 +1,646 @@
+		messageListeners: make(map[string]*messageListener),
+	}
+	gm.messagesInit()
+	gm.rpcEventStreams = newRPCEventStreams(gm)
+	gm.bgCtx, gm.cancelCtx = context.WithCancel(bgCtx)
+	return gm
+}
+
+func (gm *groupManager) PreInit(pic components.PreInitComponents) (*components.ManagerInitResult, error) {
+	gm.initRPC()
+	return &components.ManagerInitResult{
+		RPCModules: []*rpcserver.RPCModule{gm.rpcModule},
+	}, nil
+}
+
+func (gm *groupManager) PostInit(c components.AllComponents) error {
+	gm.stateManager = c.StateManager()
+	gm.txManager = c.TxManager()
+	gm.domainManager = c.DomainManager()
+	gm.p = c.Persistence()
+	gm.transportManager = c.TransportManager()
+	gm.registryManager = c.RegistryManager()
+	return gm.loadMessageListeners()
+}
+
+func (gm *groupManager) Start() error {
+	gm.startMessageListeners()
+	return nil
+}
+
+func (gm *groupManager) Stop() {
+	gm.rpcEventStreams.stop()
+	gm.stopMessageListeners()
+	gm.cancelCtx()
+}
+
+func (gm *groupManager) validateProperties(ctx context.Context, spec *pldapi.PrivacyGroupInput) (err error) {
+
+	// If no specific ABI has been provided for the input properties, we infer one from the properties
+	// that have been supplied. For example if `{"name": "group1", "transaction": "dvp-12345"}` is supplied in the properties, we would
+	// infer a properties ABI spec of:
+	//
+	//  [ { "name": "name", "type": "string", "indexed": true }, { "name": "transaction", "type": "string", "indexed": true } ]
+	//
+	// Noting details in the ABIInferenceFromJSON() implementation on constraints, and field ordering etc.
+	schemaGenerated := false
+	if spec.PropertiesABI == nil {
+		schemaGenerated = true
+		spec.PropertiesABI, err = tktypes.ABIInferenceFromJSON(ctx, spec.Properties)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Now we do standardized formatting of the properties back again into the data we pass to the domain.
+	// This validates that edge cases are handled (such as mixed arrays in the input properties, which are not supported with ABI inference).
+	// It also standardizes the structure of the data format that is passed to the domain code, prior to the full state ABI/data being stored.
+	if spec.Properties == nil {
+		spec.Properties = tktypes.RawJSON(`{}`)
+	}
+	cv, err := spec.PropertiesABI.ParseJSONCtx(ctx, spec.Properties)
+	if err == nil {
+		spec.Properties, err = tktypes.StandardABISerializer().SerializeJSONCtx(ctx, cv)
+	}
+	if err != nil {
+		return i18n.WrapError(ctx, err, msgs.MsgPGroupsDataDoesNotMatchSchema, schemaGenerated)
+	}
+
+	return nil
+
+}
+
+func (gm *groupManager) validateMembers(ctx context.Context, members []string) (remoteMembers map[string][]string, err error) {
+	localNode := gm.transportManager.LocalNodeName()
+	remoteMembers = make(map[string][]string)
+	if len(members) == 0 {
+		return nil, i18n.NewError(ctx, msgs.MsgPGroupsNoMembers)
+	}
+	for _, m := range members {
+		_, node, err := tktypes.PrivateIdentityLocator(m).Validate(ctx, "", false)
```

Comment:
Does this indicate that all member names must be fully qualified with node names?

Reply:
Yes. I've updated the field docs to make that clear.